src/client_side.cc

```diff
@@ -3020,15 +3025,23 @@ ConnStateData::parseTlsHandshake()
     // We should disable read/write handlers
     Comm::ResetSelect(clientConnection->fd);
 
-    if (unsupportedProtocol) {
+    if (parseErrorDetails) {
         Http::StreamPointer context = pipeline.front();
         Must(context && context->http);
         HttpRequest::Pointer request = context->http->request;
         debugs(83, 5, "Got something other than TLS Client Hello. Cannot SslBump.");
-        sslBumpMode = Ssl::bumpSplice;
-        context->http->al->ssl.bumpMode = Ssl::bumpSplice;
+        updateError(ERR_PROTOCOL_UNKNOWN, parseErrorDetails);
         if (!clientTunnelOnError(this, context, request, HttpRequestMethod(), ERR_PROTOCOL_UNKNOWN))
             clientConnection->close();
+        else {
+            // XXX: Why do we need to set these? The tunnels initiated by
+            // clientTunnelOnError() seem a bit different than splicing tunnels;
+            // for example, they work for configurations with disabled SslBump.
+            // And if all SslBump-related callers should set these, then perhaps
+            // clientTunnelOnError() or the code it calls can do that instead?
+            sslBumpMode = Ssl::bumpSplice;
+            context->http->al->ssl.bumpMode = Ssl::bumpSplice;
+        }
```

Comment:
There is not any need to set sslBumpMode or al::ssl:bumpMode to tunnel.
I suppose they are set because of logging purposes. The connection is under the sslBumping rules and finally spliced because it was not TLS.  
I am not finding bad (if the exact error/reason is logged with a way). Else we can comment out and use the "bumpErr" state in BumpMode enum (ssl/support.h file)

Reply:
Thank you for checking this. I removed sslBumpMode reset in branch commit dafb98b.